### PR TITLE
R4 Financial transaction guarantor balance account changes

### PR DESCRIPTION
**Pull request by Anurag-Balannagari**
Description
----
Financial transaction guarantor balance account changes
Screen Shot 2022-01-24 at 1 14 43 PM
Screen Shot 2022-01-21 at 11 41 25 AM
Screen Shot 2022-01-21 at 11 42 01 AM
Screen Shot 2022-01-21 at 11 42 24 AM
Screen Shot 2022-01-21 at 11 42 47 AM

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: lib/resources/r4/financial_transaction.yaml [hunk 1]
```diff
@@ -248,9 +247,12 @@ fields:
...
     type: Extension
     description: The specific target that the payment or adjustment should be posted to. This extension is nested under a <code>financial-transaction-allocation</code> extension.
-    note:
-      - The <code>valueReference.reference</code> must be an Encounter reference.
-      - The URL for this extension is <code>target</code>.
+    note: >
+      <ul>
+        <li>The <code>valueReference.reference</code> must be either an Encounter or Account reference.</li>
+        <li>The URL for this extension is <code>target</code>.</li>
+        <li>The Account reference must be of the guarantor-balance type.</li>
```
[the remark this answers is not in the document]
Comment (Anurag-Balannagari) [reply lib/resources/r4/financial_transaction.yaml:253]: updated here https://github.com/cerner/fhir.cerner.com/pull/794/commits/a5ff8fc50b05d7b080fb1db4488bcfc0c4788425#diff-fbd410f5e96863ae93ac388f712817417b0247695fdb087032c433dd619b4239R253

**Review by alexbeckner1 — APPROVED** [review]

File: lib/resources/r4/financial_transaction.yaml [hunk 2]
```diff
@@ -248,9 +247,12 @@ fields:
     required: 'Yes'
     type: Extension
     description: The specific target that the payment or adjustment should be posted to. This extension is nested under a <code>financial-transaction-allocation</code> extension.
-    note:
-      - The <code>valueReference.reference</code> must be an Encounter reference.
-      - The URL for this extension is <code>target</code>.
+    note: >
+      <ul>
+        <li>The <code>valueReference.reference</code> must be either an Encounter or Account reference.</li>
+        <li>The Account reference must be of the guarantor-balance type.</li>
```
Comment (Coffeevee) [inline lib/resources/r4/financial_transaction.yaml:253]: I think it reads better to say: `The Account reference must be of the <code>guarantor-balance</code> type.`

Comment (exopsRamba) [reply lib/resources/r4/financial_transaction.yaml:253]: Let's also add a conditional, i.e.
```
If there reference is of type Account, it must be a guarantor-balance account type.
```

Comment (Anurag-Balannagari) [reply lib/resources/r4/financial_transaction.yaml]: updated here https://github.com/cerner/fhir.cerner.com/pull/794/commits/4ca97ce64ea00939f679e4d26c2073012dab3271#diff-fbd410f5e96863ae93ac388f712817417b0247695fdb087032c433dd619b4239R253

**Comment by Anurag-Balannagari** [comment]
> let's update the first screenshot after Alex's feedback

updated

File: lib/resources/r4/financial_transaction.yaml (hunk 2 above)
Comment (Coffeevee) [reply lib/resources/r4/financial_transaction.yaml]: Left a nit on the commit - https://github.com/cerner/fhir.cerner.com/commit/4ca97ce64ea00939f679e4d26c2073012dab3271#r64522308

Comment (Coffeevee) [reply lib/resources/r4/financial_transaction.yaml]: Reminder to update the screenshot after making the nit update as well :)

Comment (Anurag-Balannagari) [reply lib/resources/r4/financial_transaction.yaml]: updated here https://github.com/cerner/fhir.cerner.com/pull/794/commits/3be844d058fffd1b72d105d9ae0aa19e7689d466#diff-fbd410f5e96863ae93ac388f712817417b0247695fdb087032c433dd619b4239R253

**Review by Coffeevee — APPROVED** [review]

**Review by exopsRamba — APPROVED** [review]

**Review by ashu-bdvt — APPROVED** [review]

**Comment by aki-mathur** [comment]
Screenshot 2022-01-27 at 6 28 55 PM

**Pull request merged**